cypress/integration/carry_okay_spec.js

```diff
@@ -93,94 +106,96 @@ describe('CarryOkay', () => {
-    cy.get('[data-cy=song-card]').should("have.length", 1)
-      .and('contain', 'Usher');
-    cy.get('[data-cy=search-bar]').clear()
-    cy.get('[data-cy=search-bar]').type('body')
-    cy.get('[data-cy=search-btn').click()
-    cy.get('[data-cy=song-card]').should('have.length', 2);
-  });
+    it('should display Song Book page components', () => {
+      cy.get('[data-cy=home-nav]').should('exist');
+      cy.get('[data-cy=my-songs-nav]').contains('My Songs');
+      cy.get('[data-cy=song-book-nav]').contains('Song Book');
+      cy.get('[data-cy=search-form]').should('exist');
+      cy.get('[data-cy=song-card]').should('have.length', 2);
+       
+      cy.fixture('/song_data.json').then((data) => {
+        data.forEach(song => {
+          cy.get('[data-cy=song-card]').contains(song.title);
+          cy.get('[data-cy=song-card]').contains(song.artist);
+          });
+        });
+      cy.fixture('/genre_data.json').then((data) => {
+        data.forEach(genre => {
+          const genres = Object.entries(genre)
+          genres.forEach(entry => {
+            if(entry.includes(true)) {
+              cy.get('[data-cy=song-card]').contains(entry[0]);
+            }
+          });
+        });
+      });
+      cy.get('[data-cy=album-img]').first().should('exist');
+      cy.get('[data-cy=song-card-btn]').first().should('exist');
+      cy.get('[data-cy=microphone-icon]').should('exist');
+      });
+
+    it('should allow user to make a search', () => {
+      cy.get('[data-cy=search-bar]').type('usher');
+      cy.get('[data-cy=search-btn').click()
+      cy.get('[data-cy=search-msg]').contains("Showing results for 'usher':");
+      cy.get('[data-cy=song-card]').should('have.length', 1)
+        .and('contain', 'Usher');
+    });
 
-  //still need this functionality
-  it('should display a helpful message if search yields no results', () => {
-    expect(true).to.equal(false);
-  });
+    it('should display a helpful message if search yields no results', () => {
+      cy.get('[data-cy=search-bar]').type('frank sinatra');
+      cy.get('[data-cy=search-btn').click();
+      cy.get('[data-cy=search-msg]').contains('No results for this search. Time to freestyle! (Or try another search 😉)');
+    });  
 
-  it('should allow a user to add a song to their library', () => {
-    cy.get('[data-cy=song-card-btn]').first().click();
-    cy.get('[data-cy=my-songs-nav]').click()
-    cy.get('[data-cy=song-card]').should("contain", "Jigsaw falling into place")
-        .and("contain", "Radiohead")
-        .and("contain","Electronica")
-        .and("contain","Rock")
+    it('should allow a user to add a song to their library', () => {
+      cy.get('[data-cy=song-card-btn]').first().click();
+      cy.get('[data-cy=my-songs-nav]').click();
+      cy.get('[data-cy=song-card]').should('contain', 'Jigsaw falling into place')
+          .and('contain', 'Radiohead')
+          .and('contain','Electronica')
+          .and('contain','Rock');
```

Comment:
small, but are we missing spacing on these two lines or the other way around?

Reply:
I noticed it was tabbed over to the right an extra space is that what you mean?

Reply:
line 170 'Radiohead' has some spacing right before it vs 171-172 don't